models/forum.js

```diff
@@ -1,17 +1,34 @@
 const mongoose = require('mongoose');
 
 const forumSchema = new mongoose.Schema({
+  parentId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Forum',
+    default: null,
+    validate: {
+      validator: function(id) {
+        if (this.isAnswer && !id) {
+          return false;
+        } else {
+          return true;
+        }
+      },
+      message: 'parentId is reuired for an answers'
+    }
+  },
   title: {
     type: String,
     required: true
   },
   text: { type: String, required: true },
+  isAnswer: { type: Boolean, required: true, default: false },
   createdAt: { type: Date, default: Date.now },
   updatedAt: { type: Date, default: Date.now },
-  poster: {
+  author: {
     type: mongoose.Schema.Types.ObjectId,
     ref: 'User'
   },
+  answers: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Forum' }],
```

Comment:
We are not having upvotes and downvotes for forum answers ?

Reply:
Yes, but now as answers are of same model as forum, same routes will work for both.

Reply:
Yup, was a problem to interact with the answers, so we're going a linked-list like approach.